frontend/src/pages/article/article-page.tsx

```diff
@@ -1,22 +1,44 @@
-import { type FC } from 'react';
+import { Layout, Navigate } from '~/libs/components/components.js';
+import { Loader } from '~/libs/components/loader/loader.js';
+import { AppRoute, DataStatus } from '~/libs/enums/enums.js';
+import { getFullName } from '~/libs/helpers/helpers.js';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useEffect,
+  useParams,
+} from '~/libs/hooks/hooks.js';
+import { type ArticleType, type TagType } from '~/libs/types/types.js';
+import { actions } from '~/slices/articles/articles.js';
 
-import { Layout } from '~/libs/components/components.js';
-import { useParams } from '~/libs/hooks/hooks.js';
-import { type TagType } from '~/libs/types/types.js';
-
-import { ArticleView, Author } from './components/components.js';
+import { ArticleView, AuthorDetails } from './components/components.js';
 import styles from './styles.module.scss';
 
-const ArticlePage: FC = () => {
+const ArticlePage: React.FC = () => {
+  const dispatch = useAppDispatch();
+
   const { id } = useParams();
 
-  const MOCKED_TEXT = `Envision this: there is a technology currently undergoing testing that, when released to the public, will become a long-awaited revolution in energy. This new technology promises to be safer and more efficient than anything we have on the market now. It  will affect that which we consider mundane — power tools, toys, laptops, smartphones —
-  and that which we consider exceptional — medical devices, spacecraft, and the innovative new vehicle designs needed to wean us off of fossil fuels. We have known about this  technology for centuries, yet until now we have only been able to take small steps towards its creation. Billions of dollars are pouring into research and billions more will be made once the
-  technology has been perfected and released.
+  useEffect(() => {
+    void dispatch(actions.getArticle(Number(id)));
+  }, [dispatch, id]);
+
+  const { article, dataStatus } = useAppSelector(({ articles }) => ({
+    article: articles.article,
+    dataStatus: articles.dataStatus,
+  }));
+
+  const isLoading = !(
+    dataStatus === DataStatus.FULFILLED || dataStatus == DataStatus.REJECTED
+  );
+
```

Comment:
Why we can't check `isLoading` like this:
```suggestion
const isLoading = dataStatus === DataStatus.PENDING;
```

Reply:
because there is an initial dataStatus - idle